homeassistant/components/asuswrt/__init__.py

```diff
@@ -77,10 +78,13 @@
         dnsmasq=conf[CONF_DNSMASQ],
     )
 
-    await api.connection.async_connect()
+    try:
+        await api.connection.async_connect()
+    except OSError as ex:
+        raise PlatformNotReady() from ex
+
     if not api.is_connected:
-        _LOGGER.error("Unable to setup component")
-        return False
+        raise PlatformNotReady()
```

Comment:
This seems redundant when the try-catch is in place! I don't think its needed at all

Reply:
I don't know what api do inside, but is different condition. 
May be we don't have an exception but the property return False. In this case or we also return false or we raise the exception. Not manage it at all seesm wrong to me.

Reply:
Its not needed imo. it can be used in runtime to determine if the connection have dropped.
```python
    async def async_connect(self):
        """Fetches the client or creates a new one."""

        kwargs = {
            'username': self._username if self._username else None,
            'client_keys': [self._ssh_key] if self._ssh_key else None,
            'port': self._port,
            'password': self._password if self._password else None,
            'known_hosts': None
        }

        self._client = await asyncssh.connect(self._host, **kwargs)
        self._connected = True
```
The `self._connected` is the value behind `self.connected property`  and it will get set in the `async_connect`

Reply:
Make sense. I removed this check

Reply:
Tests now are failing, I think I have to revert....

Reply:
`PlatformNotReady` doesn't work in `async_setup`

Reply:
So, there are alternative solution?

Reply:
Convert the integration to a config flow and use `ConfigEntryNotReady` in `async_setup_entry`.

Reply:
More long story. If this is the only solution I have to close this PR for the moment.

Reply:
The other alternative is to catch the exception and create your own retry mechanism with `async_call_later` from helpers.event.

Reply:
This last commit work for me. Please let me know...